internal/loader/typed_setting.go

```diff
@@ -118,3 +104,51 @@ func isJsonContentType(contentType *string) bool {
 	matched, _ := regexp.MatchString("^application\\/(?:[^\\/]+\\+)?json(;.*)?$", contentTypeStr)
 	return matched
 }
+
+// UnmarshalConfigMap is used for scenarios related to feature flag refresh.
+// Currently, only supports feature flag settings in JSON/YAML formats.
+func UnmarshalConfigMap(existingConfigMapSetting *map[string]string, dataOptions *acpv1.ConfigMapDataOptions) (map[string]interface{}, map[string]interface{}, error) {
+	allSettings := make(map[string]interface{})
+	existingConfigMapData := (*existingConfigMapSetting)[dataOptions.Key]
```

Comment:
This method `UnmarshalConfigMap` should only be called during refreshing process, so I know there should be no reason that `(*existingConfigMapSetting)[dataOptions.Key]` does not exists, hence we'd better handle it by throwing an exception if `dataOptions.Key` does not exist, instead of a silent error